geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/blocks/BlockTracking.kt

```diff
@@ -0,0 +1,95 @@
+package com.mineinabyss.geary.papermc.tracking.blocks
+
+import com.mineinabyss.geary.addons.dsl.GearyAddonWithDefault
+import com.mineinabyss.geary.modules.geary
+import com.mineinabyss.geary.papermc.tracking.blocks.helpers.GearyBlockPrefabQuery
+import com.mineinabyss.geary.papermc.tracking.blocks.systems.TrackOnSetBlockComponent
+import com.mineinabyss.geary.prefabs.PrefabKey
+import com.mineinabyss.idofront.di.DI
+import org.bukkit.Instrument
+import org.bukkit.Material
+import org.bukkit.Note
+import org.bukkit.block.BlockFace
+import org.bukkit.block.data.BlockData
+import org.bukkit.block.data.type.CaveVines
+import org.bukkit.block.data.type.NoteBlock
+import org.bukkit.block.data.type.Tripwire
+
+val gearyBlocks by DI.observe<BlockTracking>()
+interface BlockTracking {
+    val blockMap: Map<BlockData, Int>
+    val blockProvider: GearyBlockProvider
+    val block2Prefab: Block2Prefab
+    val blockPrefabs: GearyBlockPrefabQuery
+
+    fun createBlockData(prefabKey: PrefabKey): BlockData? = blockProvider.serializePrefabToBlockData(prefabKey)
+
+    companion object : GearyAddonWithDefault<BlockTracking> {
+        override fun default(): BlockTracking = object : BlockTracking {
+            override val blockMap = createBlockMap()
+            override val blockProvider = GearyBlockProvider()
+            override val blockPrefabs = GearyBlockPrefabQuery()
+            override val block2Prefab = Block2Prefab()
+        }
+
+        override fun BlockTracking.install() {
+            geary.pipeline.addSystems(
+                TrackOnSetBlockComponent()
+            )
+        }
+
+        private fun createBlockMap(): MutableMap<BlockData, Int> {
```

Comment:
Function for filling out the blockMap

Reply:
This might be worth moving into a separate class so that we can mess with the implementation for tests (or multi version support if we're ever going to do it), as it stands right now you can't change what the generated block map becomes if needed.

Maybe Block2Prefab could work, also see my comment towards the end about changing the structure a bit